src/main/java/uk/gov/hmcts/ccd/auditlog/aop/ExpressionEvaluator.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.ccd.auditlog.aop;
+
+import org.springframework.aop.support.AopUtils;
+import org.springframework.context.expression.AnnotatedElementKey;
+import org.springframework.context.expression.CachedExpressionEvaluator;
+import org.springframework.context.expression.MethodBasedEvaluationContext;
+import org.springframework.core.DefaultParameterNameDiscoverer;
+import org.springframework.core.ParameterNameDiscoverer;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ *  Utility class handling the SpEL expression parsing
+ *
+ */
+public class ExpressionEvaluator extends CachedExpressionEvaluator {
+
+    // shared param discoverer since it caches data internally
+    private final ParameterNameDiscoverer paramNameDiscoverer = new DefaultParameterNameDiscoverer();
+
+    private final Map<ExpressionKey, Expression> conditionCache = new ConcurrentHashMap<>(64);
+
+    private final Map<AnnotatedElementKey, Method> targetMethodCache = new ConcurrentHashMap<>(64);
+
+    public EvaluationContext createEvaluationContext(Object object, Class<?> targetClass, Method method, Object[] args) {
+        Method targetMethod = getTargetMethod(targetClass, method);
+        ExpressionRootObject root = new ExpressionRootObject(object, args);
+        return new MethodBasedEvaluationContext(root, targetMethod, args, this.paramNameDiscoverer);
+    }
+
+    public <T> T condition(String conditionExpression, AnnotatedElementKey elementKey, EvaluationContext evalContext, Class<T> clazz) {
+        return getExpression(this.conditionCache, elementKey, conditionExpression).getValue(evalContext, clazz);
+    }
+
+    private Method getTargetMethod(Class<?> targetClass, Method method) {
+        AnnotatedElementKey methodKey = new AnnotatedElementKey(method, targetClass);
+        Method targetMethod = this.targetMethodCache.get(methodKey);
+        if (targetMethod == null) {
+            targetMethod = AopUtils.getMostSpecificMethod(method, targetClass);
+            this.targetMethodCache.put(methodKey, targetMethod);
+        }
+        return targetMethod;
+    }
+
+    private static class ExpressionRootObject {
+        private final Object object;
+        private final Object[] args;
+
+        public ExpressionRootObject(Object object, Object[] args) {
+            this.object = object;
+            this.args = args;
+        }
+
+        public Object getObject() {
+            return object;
+        }
+
+        public Object[] getArgs() {
+            return args;
+        }
+    }
```

Comment:
public Object[] getArgs() {. }.  It is not used anywhere and it is not overwriting .

Reply:
That class similar to VO/DTO.

Reply:
It is ok . there has to be a getter.